File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.java
class Solution {
    public boolean canBeEqual(int[] a, int[] b) {
         int m = a.length, n = b.length;
        //if (m != n) return false;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < m; i++) {
            map.put(a[i], map.getOrDefault(a[i], 0) + 1);
            map.put(b[i], map.getOrDefault(b[i], 0) - 1);
        }
        for (int i : map.keySet()) {
            if (map.get(i) != 0) return false;
        }
        return true;
    }
}

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        unordered_map<int, int> ump;
    for(int itr = 0; itr < target.size(); itr++){
        ump[target[itr]]++;
        ump[arr[itr]]--;
    }        
    for(int itr = 0; itr < ump.size(); itr++){
        if(ump[itr] != 0) return false;
    }
    return true;
    }
};